sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -288,6 +297,65 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext
     }
   }
 
+  test("SQL interface support storageLevel(DISK_ONLY)") {
```

Comment:
@wangyum . Could you refactor the following four positive test cases by reducing the duplication? They are mostly the same except SQL statements and `StorageLevel`s.
```scala
test("SQL interface support storageLevel(DISK_ONLY)")
test("SQL interface select from table support storageLevel(DISK_ONLY)")
test("SQL interface support storageLevel(DISK_ONLY) with invalid options")
test("SQL interface support storageLevel(MEMORY_ONLY)")
```

Reply:
How about this:
```scala
Seq("LAZY", "").foreach { isLazy =>
  Seq(true, false).foreach { withInvalidOptions =>
    Seq(true, false).foreach { withCacheTempView =>
      Map("DISK_ONLY" -> Disk, "MEMORY_ONLY" -> Memory).foreach {
        case (storageLevel, dataReadMethod) =>
          val testName = s"SQL interface support option: storageLevel: $storageLevel, " +
            s"isLazy: ${isLazy.equals("LAZY")}, " +
            s"withInvalidOptions: $withInvalidOptions, withCacheTempView: $withCacheTempView"
          val cacheOption = if (withInvalidOptions) {
            s"OPTIONS('storageLevel' '$storageLevel', 'a' '1', 'b' '2')"
          } else {
            s"OPTIONS('storageLevel' '$storageLevel')"
          }
          test(testName) {
            if (withCacheTempView) {
              withTempView("testSelect") {
                sql(s"CACHE $isLazy TABLE testSelect $cacheOption SELECT * FROM testData")
                assertCached(spark.table("testSelect"))
                val rddId = rddIdOf("testSelect")
                if (isLazy.equals("LAZY")) {
                  sql("SELECT COUNT(*) FROM testSelect").collect()
                }
                assert(isExpectStorageLevel(rddId, dataReadMethod))
              }
            } else {
              sql(s"CACHE $isLazy TABLE testData $cacheOption")
              assertCached(spark.table("testData"))
              val rddId = rddIdOf("testData")
              if (isLazy.equals("LAZY")) {
                sql("SELECT COUNT(*) FROM testData").collect()
              }
              assert(isExpectStorageLevel(rddId, dataReadMethod))
            }
          }
      }
    }
  }
}
```

Reply:
@dongjoon-hyun  Such test cases don't seem to be clear. Can we use the original 4 test cases?

Reply:
@wangyum . Originally, I thought you can make a utility function for those 4 test cases. 
The utility function can accept `sql` statement and `storagelevel`. Then, 4 test cases will have simpler body.
> They are mostly the same except SQL statements and StorageLevels.

Reply:
Thanks @dongjoon-hyun  How about:
```scala

  def assertStorageLevel(
      sqlStr: String,
      tableName: String,
      level: DataReadMethod.DataReadMethod): Unit = {
    sql(sqlStr)
    assertCached(spark.table(tableName))
    val rddId = rddIdOf(tableName)
    assert(isExpectStorageLevel(rddId, level))
  }

  test("SQL interface support storageLevel(DISK_ONLY)") {
    assertStorageLevel(
      "CACHE TABLE testData OPTIONS('storageLevel' 'DISK_ONLY')",
      "testData",
      Disk)
  }

  test("SQL interface cache SELECT ... support storageLevel(DISK_ONLY)") {
    withTempView("testCacheSelect") {
      assertStorageLevel(
        "CACHE TABLE testCacheSelect OPTIONS('storageLevel' 'DISK_ONLY') SELECT * FROM testData",
        "testCacheSelect",
        Disk)
    }
  }

  test("SQL interface support storageLevel(DISK_ONLY) with invalid options") {
    assertStorageLevel(
      "CACHE TABLE testData OPTIONS('storageLevel' 'DISK_ONLY', 'a' '1', 'b' '2')",
      "testData",
      Disk)
  }

  test("SQL interface support storageLevel(MEMORY_ONLY)") {
    assertStorageLevel(
      "CACHE TABLE testData OPTIONS('storageLevel' 'MEMORY_ONLY')",
      "testData",
      Memory)
  }
```

Reply:
Can we do the following? Then, those 4 function will have one-line body.
1. Let's remove `tableName: String,`. We can use the same table name.
2. With (1), we can remove the common prefix, `CACHE TABLE testData OPTIONS(`, too.
3. With (1) and (2), we can change `sqlStr: String` -> `options: String`.